doc/man3/EVP_DigestSignInit.pod

```diff
@@ -26,7 +26,25 @@ EVP_DigestSignInit() sets up signing context B<ctx> to use digest B<type> from
 ENGINE B<e> and private key B<pkey>. B<ctx> must be created with
 EVP_MD_CTX_new() before calling this function. If B<pctx> is not NULL the
 EVP_PKEY_CTX of the signing operation will be written to B<*pctx>: this can
-be used to set alternative signing options.
+be used to set alternative signing options. The digest B<type> may be NULL if
+the signing algorithm supports it.
+
+Only EVP_PKEY types that support signing can be used with these functions. This
+includes MAC algorithms where the MAC generation is considered as a form of
+"signing". Built-in EVP_PKEY types supported by these functions are CMAC,
+Poly1305, DSA, HMAC, RSA, Siphash, Ed25519 and Ed448.
+
+Not all digests can be used for all key types. The following combinations apply.
+DSA supports SHA1, SHA224, SHA256, SHA384 and SHA512. ECDSA supports SHA1,
```

Comment:
Would this be more digestible as a table or list?  (E.g., using over/item/back constructs). At a minimum, it should be alphabetized.

Reply:
Ok, done.